### PR TITLE
esp-js-react-agile-board updates (typescript, vdom optimisations)

### PR DESCRIPTION
**Pull request by KeithWoods**

### CODE REVIEW

File: examples/api/README.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ This will install all dependencies.
 
 `npm start`
 
-This will call babel to transpile app.js and run it via node. 
+This will call babel to transpile app.tsx and run it via node.
```
Comment (raybooysen) [inline examples/api/README.md]: Babel has been removed.

File: examples/esp-js-react-agile-board/package.json [hunk 2]
```diff
@@ -9,25 +9,24 @@
...
-    "babel-preset-stage-0": "^6.3.13",
-    "webpack": "^1.12.12",
-    "webpack-dev-server": "^1.14.1"
+    "@types/react": "16.0.9",
+    "clean-webpack-plugin": "^0.1.16",
+    "tslint": "5.8.0",
+    "tslint-loader": "3.5.3",
+    "typescript": "^2.2.2",
+    "webpack": "^3.10.0",
+    "webpack-fail-plugin": "^2.0.0"
```
Comment (raybooysen) [inline examples/esp-js-react-agile-board/package.json]: You no longer need the fail plugin with webpack 3 and up

File: examples/esp-js-react-agile-board/src/eventConsts.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+export class EventConsts {
```
Comment (raybooysen) [inline examples/esp-js-react-agile-board/src/eventConsts.ts]: don't need a class. :)

File: examples/esp-js-react-agile-board/tslint.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
+{
```
Comment (raybooysen) [inline examples/esp-js-react-agile-board/tslint.json]: tslint walks the tree, so you can move this right to the root and all projects will use it.

Comment (KeithWoods) [reply examples/esp-js-react-agile-board/tslint.json]: This is the only place that has typescript ATM

File: examples/api/README.md (hunk 1 above)
Comment (KeithWoods) [reply examples/api/README.md]: oops, this example actually still using it. I must have picked up this ext in a search and replace. I only updated the main example (esp-js-react-agile-board) to TypeScript.

**Pull request merged**